### ISSUE TITLE
Unable to install mantid using conda

### ISSUE THREAD

**Issue by beenje** [mantidproject/conda-recipes#18]
I'm trying to install mantid using conda but it fails with the following error:
`ValidationError: Value for sha256 cannot be None.`

It does say that it's a conda error but as I only saw this error when trying to install mantid, I thought I'd ask here first.

Here is the full log:
```
$ conda create -n mantid -c mantid mantid-framework
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment /opt/conda/envs/mantid:

The following NEW packages will be INSTALLED:

    boost:            1.61.0-py27_1        conda-forge
    bzip2:            1.0.6-1              conda-forge
    ca-certificates:  2017.1.23-1          conda-forge
    certifi:          2017.4.17-py27_0     conda-forge
    cmake:            3.7.1-4              conda-forge
    curl:             7.52.1-0             conda-forge
    dateutil:         2.4.1-py27_0
    expat:            2.1.0-2              conda-forge
    gsl:              2.2.1-1              conda-forge
    h5py:             2.6.0-np111py27_3    conda-forge
    hdf4:             4.2.12-0             conda-forge
    hdf5:             1.8.15.1-3
    icu:              56.1-4               conda-forge
    jpeg:             9b-0                 conda-forge
    jsoncpp:          0.10.6-1             conda-forge
    libarchive:       3.2.1-2              conda-forge
    libgcc:           5.2.0-0
    libgfortran:      3.0.0-1
    libiconv:         1.14-4               conda-forge
    libuv:            1.11.0-0             conda-forge
    mantid-framework: 3.9.2unstable-py27_0 mantid
    mkl:              2017.0.1-0
    mpi4py:           2.0.0-py27_2         conda-forge
    mpich:            3.2-4                conda-forge
    muparser:         2.2.5-2              conda-forge
    ncurses:          5.9-10               conda-forge
    nexus:            4.3.1-py27_0         mantid
    numpy:            1.11.3-py27_0
    openblas:         0.2.19-2             conda-forge
    openssl:          1.0.2k-0             conda-forge
    pip:              9.0.1-py27_0         conda-forge
    poco:             1.7.3-py27_0         mantid
    python:           2.7.13-0             conda-forge
    readline:         6.2-0                conda-forge
    scipy:            0.19.0-np111py27_0
    setuptools:       33.1.1-py27_0        conda-forge
    six:              1.10.0-py27_1        conda-forge
    sqlite:           3.13.0-1             conda-forge
    tbb:              2017_20170226-0      conda-forge
    tk:               8.5.19-1             conda-forge
    wheel:            0.29.0-py27_0        conda-forge
    xz:               5.2.2-0              conda-forge
    zlib:             1.2.11-0             conda-forge

Proceed ([y]/n)? y

An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : linux-64
          conda version : 4.3.11
       conda is private : False
      conda-env version : 4.3.11
    conda-build version : not installed
         python version : 2.7.13.final.0
       requests version : 2.12.4
       root environment : /opt/conda  (writable)
    default environment : /opt/conda
       envs directories : /opt/conda/envs
                          /home/conda/.conda/envs
          package cache : /opt/conda/pkgs
                          /home/conda/.conda/pkgs
           channel URLs : https://conda.anaconda.org/mantid/linux-64
                          https://conda.anaconda.org/mantid/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /home/conda/.condarc
           offline mode : False
             user-agent : conda/4.3.11 requests/2.12.4 CPython/2.7.13 Linux/4.9.13-moby CentOS Linux/7.3.1611 glibc/2.17
                UID:GID : 998:100



`$ /opt/conda/bin/conda create -n mantid -c mantid mantid-framework`




    Traceback (most recent call last):
      File "/opt/conda/lib/python2.7/site-packages/conda/exceptions.py", line 616, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/opt/conda/lib/python2.7/site-packages/conda/cli/main.py", line 137, in _main
        exit_code = args.func(args, p)
      File "/opt/conda/lib/python2.7/site-packages/conda/cli/main_create.py", line 68, in execute
        install(args, parser, 'create')
      File "/opt/conda/lib/python2.7/site-packages/conda/cli/install.py", line 359, in install
        execute_actions(actions, index, verbose=not context.quiet)
      File "/opt/conda/lib/python2.7/site-packages/conda/plan.py", line 825, in execute_actions
        execute_instructions(plan, index, verbose)
      File "/opt/conda/lib/python2.7/site-packages/conda/instructions.py", line 258, in execute_instructions
        cmd(state, arg)
      File "/opt/conda/lib/python2.7/site-packages/conda/instructions.py", line 118, in UNLINKLINKTRANSACTION_CMD
        txn = UnlinkLinkTransaction.create_from_dists(index, prefix, unlink_dists, link_dists)
      File "/opt/conda/lib/python2.7/site-packages/conda/core/link.py", line 123, in create_from_dists
        for dist, pkg_dir in zip(link_dists, pkg_dirs_to_link))
      File "/opt/conda/lib/python2.7/site-packages/conda/core/link.py", line 123, in <genexpr>
        for dist, pkg_dir in zip(link_dists, pkg_dirs_to_link))
      File "/opt/conda/lib/python2.7/site-packages/conda/gateways/disk/read.py", line 88, in read_package_info
        paths_data = read_paths_json(extracted_package_directory)
      File "/opt/conda/lib/python2.7/site-packages/conda/gateways/disk/read.py", line 150, in read_paths_json
        paths=(PathDataV1(**f) for f in data['paths']),
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 702, in __call__
        instance = super(EntityType, cls).__call__(*args, **kwargs)
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 719, in __init__
        setattr(self, key, kwargs[key])
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 424, in __set__
        instance.__dict__[self.name] = self.validate(instance, self.box(instance, val))
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 581, in box
        return self._type(v if isinstance(v, et) else et(**v) for v in val)
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 581, in <genexpr>
        return self._type(v if isinstance(v, et) else et(**v) for v in val)
      File "/opt/conda/lib/python2.7/site-packages/conda/gateways/disk/read.py", line 150, in <genexpr>
        paths=(PathDataV1(**f) for f in data['paths']),
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 702, in __call__
        instance = super(EntityType, cls).__call__(*args, **kwargs)
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 719, in __init__
        setattr(self, key, kwargs[key])
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 424, in __set__
        instance.__dict__[self.name] = self.validate(instance, self.box(instance, val))
      File "/opt/conda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 465, in validate
        raise ValidationError(getattr(self, 'name', 'undefined name'), val)
    ValidationError: Value for sha256 cannot be None.
```

**Comment by yxqd** [comment]
Did you do this under a docker image?

**Comment by beenje** [comment]
Yes, I'm running that in a docker image (centos:7)

**Comment by yxqd** [comment]
You will need to install a few other packages for conda to work I think. I saw similar problem running conda directly on official ubuntu docker image. I googled and this might give hints on what to install for conda to work well: https://hub.docker.com/r/tylerthetyrant/miniconda3-centos7/~/dockerfile/

**Comment by beenje** [comment]
I've installed many packages using conda without problem in the same Docker container (pandas, sympy, scipy...).
But I tried the tylerthetyrant/miniconda3-centos7 image and it worked! I didn't see any differences with my image except the conda version used.

After updating conda from 4.3.11 to 4.3.14, the problem disappeared :-)
`conda install -y conda=4.3.14`

`conda create -n mantid -c mantid mantid-framework` worked fine.

It was a conda problem. I'll close this issue.

Thanks for pointing me to that image!
